### PR TITLE
Add warning for -flto flag

### PR DESCRIPTION
**Pull request by dead-tech**
I think this will do it for #3111.

### CODE REVIEW

File: static/panes/compiler.js
Comment (partouf) [inline static/panes/compiler.js:1958]: would be better to have an else-if and the msg inside the other if

Comment (dead-tech) [reply static/panes/compiler.js:1958]: Correct me if I'm wrong, but changing it to an `else if` would exclude the scenario where both of the branches should evalute to true (maybe this is intentional), as in the case of an `unwiseOptions` (e.g. -march=native) and also -flto being used.

As regards the variable scoping I changed that, and I also noticed that we should raise this warning only if Compile to Binary is actually turned off, therefore I've added this extra condition.

Comment (mattgodbolt) [reply static/panes/compiler.js:1958]: I'm not quite sure what our alert system does with two notifies at once; but in the interests of getting this merged I think this is just fine! Thanks for your patience!

**Review by mattgodbolt — APPROVED** [review]

File: static/panes/compiler.js
Comment (partouf) [reply static/panes/compiler.js:1958]: Lost track of this comment, but yeah that was my exact concern. I guess we will find out :)

Comment (dead-tech) [reply static/panes/compiler.js:1958]: Sorry for bothering again, and I might sound a little repetitive, but I wanted to underline the fact that at the moment the message is shown also with Compile to Binary turned on. As i mentioned earlier i had already fixed this and i was waiting for @partouf's approval to push the changes. Now since this has been already merged if it is fine for you, it's obviously fine for me, I just wanted you to be aware of it.

Comment (partouf) [reply static/panes/compiler.js:1958]: I see live that theres only 1 message at the time active, -march has preference because it's the last to be notified about, if you remove march, the hint about -flto appears. I guess it's the collapseSimilar that arranges that only the last message will show. Not the clearest from the code, but it works.

**Pull request merged**
